src/util.c

```diff
@@ -951,3 +951,12 @@ void __stack_chk_fail(void)
 
 #endif // __amigaos__
 
+#if defined(CONFIG_PSP) || defined(CONFIG_NDS)
+FILE *popen(const char *command, const char *type) {
```

Comment:
Put it here as (a) those functions are used by libxmp, so a mere #define won't suffice (unless we patch libxmp again), and (b) the Wii and 3DS also need this.